flybywiresim/aircraft
# feat(a380/fws): TCAS STBY & Fault messages

Fixes #[issue_no]

## Summary of Changes

Implements TCAS FAULT & STBY abnormal sensed ecam warning.
Disables TCAS selection on MFD if TCAS is faulty or XPDR is STBY.
Minor fixes to tcas such as verifying fault condition before transponder state, power logic & usage of IR data for true heading. 
## Screenshots (if necessary)



## References

FCOM https://youtu.be/GspxOmtreIQ?t=44



## Additional context

Logic has been added for independent tcas system faults, but since only one exists at the moment, only the TCAS 1+2 FAULT will be triggerable.

Discord username (if different from GitHub):
bruno_pt99
## Testing instructions

1) Start the airplane in cold and dark/engines off & IRS off and observe TCAS flags visible on PFD & ND.
2) No TCAS FAULT messages should be present at this point in the ecam.
3) Set the ADIRS to align, once IR1 has a valid heading the TCAS flags disappear. Also verify that the IR IN ALIGN memo appears on ecam while IRS are in align.
4) Verify on the MFD SURV page that TCAS inputs are not selectable as long as TCAS is failed or XPDR is in standby.
5) Start the engines and turn IR1 off (via pushbutton for e.g.), SURV TCAS 1+2 FAULT should appear on ECAM with an associated master caution, then, flick the ATT selector to CPT 3 and the caution should disappear alongside the TCAS flags on the displays.
5) Perform a normal takeoff and don't turn on the TCAS, a TCAS STBY master caution should happen shortly after climb thrust.
6) Verify TCAS STBY memo is visible in flight & during taxi if tcas is set to standby mode. 
7) Verify TCAS works normally

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
